### PR TITLE
Update events to enable TWAB tracking on subgraphs

### PR DESCRIPTION
**Pull request by dylandesrosier**

### CODE REVIEW

**Review by asselstine — CHANGES_REQUESTED** [review]
Need both of `amount` and `delegatedAmount`

File: test/contracts/mocks/TwabLibMock.sol [hunk 1]
```diff
@@ -13,64 +13,62 @@ contract TwabLibMock {
...
+      account.details
+    );
+    return (index, twab);
   }
 
-  function getBalanceAt(uint32 _targetTime) external view returns (uint256 balance) {
-    balance = TwabLib.getBalanceAt(account.twabs, account.details, _targetTime);
+  function getBalanceAt(uint32 _targetTime) external view returns (uint256) {
+    uint256 balance = TwabLib.getBalanceAt(account.twabs, account.details, _targetTime);
+    return balance;
```
Comment (PierrickGT) [inline test/contracts/mocks/TwabLibMock.sol:71]:
```suggestion
    return TwabLib.getBalanceAt(account.twabs, account.details, _targetTime);
```

**Review by PierrickGT — APPROVED** [review]

File: test/contracts/mocks/TwabLibMock.sol (hunk 1 above)
Comment (dylandesrosier) [reply test/contracts/mocks/TwabLibMock.sol:71]: <https://github.com/foundry-rs/foundry/pull/3128#issuecomment-1241245086>
You need to explicitly store it in a variable for it to be picked up in coverage checks

Comment (PierrickGT) [reply test/contracts/mocks/TwabLibMock.sol:71]: Ah ok. Not the best to save on gas though.

Comment (dylandesrosier) [reply test/contracts/mocks/TwabLibMock.sol:71]: It's just the mock contract. It's only used for testing!

**Pull request merged**